byterun/memory.c

```diff
@@ -452,6 +468,14 @@ CAMLexport value caml_alloc_shr (mlsize_t wosize, tag_t tag)
   return Val_hp (hp);
 }
 
+CAMLexport value caml_alloc_shr_no_raise (mlsize_t wosize, tag_t tag) {
+  value res;
+  caml_alloc_shr_return = 1;
+  res = caml_alloc_shr (wosize, tag);
+  caml_alloc_shr_return = 0;
```

Comment:
do we know for sure that `caml_alloc_shr_return` is always `0` when this function is called? Otherwise it would make sense to save the pre-call value and restore it, instead of always setting it to `0`.